src/test/java/com/microsoft/sqlserver/jdbc/KerberosTest.java

```diff
@@ -75,6 +76,30 @@ public void testUseDefaultJaasConfigConnectionStringPropertyFalse() throws Excep
         }
     }
 
+    @Tag(Constants.Kerberos)
+    @Test
+    public void testUseDefaultNativeGSSCredential() throws Exception {
+        // This is a negative test. This test should fail as expected as the JVM arg "-Dsun.security.jgss.native=true"
+        // isn't provided.
+        String connectionString = connectionStringKerberos + ";useDefaultGSSCredential=true;";
+
+        try (SQLServerConnection conn = (SQLServerConnection) DriverManager.getConnection(connectionString)) {
+            Assertions.fail(TestResource.getResource("R_expectedExceptionNotThrown"));
+        } catch (SQLServerException e) {
+            Assertions.assertEquals(e.getCause().getMessage(), TestResource.getResource("R_kerberosNativeGSSFailure"));
+        }
+    }
+
+    @Tag(Constants.Kerberos)
+    @Test
+    public void testBasicKerberosAuth() throws Exception {
+        try (SQLServerConnection conn = (SQLServerConnection) DriverManager.getConnection(connectionStringKerberos)) {
+            ResultSet rs = conn.createStatement().executeQuery(authSchemeQuery);
+            rs.next();
+            Assertions.assertEquals(kerberosAuth, rs.getString(1));
+        }
+    }
+
     /**
      * Overwrites the default JAAS config. Call before making a connection.
      */
```

Comment:
can we add a bit more description about this kafka stuff what's being done here

Reply:
There's nothing Kafka related going on here. In the function, the most important line is line 111 where the JAAS config is being overwritten eg. `Configuration.setConfiguration(new InternalConfiguration(configurationEntries))`. Everything else is just setup for line 111.

I'll remove the `kafka` in the context name.